src/init.js

```diff
@@ -1,6 +1,19 @@
 /* global wikipediapreview_init_options */
-wikipediaPreview.init( {
-	root: document,
-	/* eslint-disable-next-line camelcase */
-	detectLinks: !! wikipediapreview_init_options.detectLinks,
+import { getColorScheme, observeDarkModePluginActivation } from './link/colorScheme';
+
+const init = ( scheme ) => {
+	wikipediaPreview.init( {
+		root: document,
+		/* eslint-disable-next-line camelcase */
+		detectLinks: !! wikipediapreview_init_options.detectLinks,
+		prefersColorScheme: scheme,
+	} );
+};
+
+// Initial init on first page load
+init( getColorScheme() );
+
+// Re-init as needed as the color scheme changes
+observeDarkModePluginActivation( ( scheme ) => {
```

Comment:
You could further simplify by passing the function directly instead of creating a lambda with the same signature as the function.